ogre/src/OgreCOMVisual.cc

```diff
@@ -49,8 +49,13 @@ OgreCOMVisual::~OgreCOMVisual()
 //////////////////////////////////////////////////
 void OgreCOMVisual::PreRender()
 {
-  if (this->dirtyCOMVisual)
+  if (this->HasParent() && this->ParentName().empty())
+    this->parentName = this->Parent()->Name();
+
+  if (this->dirtyCOMVisual &&
+      !this->ParentName().empty())
   {
+    this->parentName = this->Parent()->Name();
```

Comment:
from the user's perspective, they can get the parent name like you did here `Parent()->Name()` so I think `ParentName()` is a duplicate public facing API that we don't need to expose. Internally in this class, we can either make `ParentName()` private or replace `ParentName()` calls with `this->parentName`

Reply:
I have replaced all calls with `this->parentName`
[`3300dc2`](https://github.com/ignitionrobotics/ign-rendering/pull/345/commits/3300dc22a3ec2fd4b7d19f4ae6f69ead7a089f47).